src/mongocxx/search_index_model.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <string>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/document/view_or_value.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+#include <bsoncxx/string/view_or_value.hpp>
+
+#include <mongocxx/config/prelude.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+///
+/// Class representing a search index on a MongoDB server.
+///
+class MONGOCXX_API search_index_model {
+   public:
+    ///
+    /// Initializes a new search_index_model over a mongocxx::collection.
+    ///
+    search_index_model(const bsoncxx::document::view_or_value& definition);
```

Comment:
```suggestion
    search_index_model(bsoncxx::document::view_or_value definition);
```
view_or_value parameters should not be `const&`.